torch/nn/functional.py

```diff
@@ -4824,12 +4824,15 @@ def _scaled_dot_product_attention(
     q = q / math.sqrt(E)
     # (B, Nt, E) x (B, E, Ns) -> (B, Nt, Ns)
     attn = torch.bmm(q, k.transpose(-2, -1))
+    if attn.isinf().any().item() and attn.dtype is torch.float16:
```

Comment:
This is introducing a sync point as well as recomputing the same thing twice.
I am not convinced this is the best approach.

cc @ngimel

Reply:
Agree with @albanD, this is a sync point and converting inputs under the hood. Properly scaled models don't result in overflow (there are many examples of transformers training with fp16), so paying this perf penalty always is not an option. cc @ptrblck, @mcarilli.